stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/preview/internal/StreamVideoPlayerContent.kt

```diff
@@ -0,0 +1,255 @@
+            context = context,
+            onBuffering = onBuffering,
+            onPlaybackError = onPlaybackError,
+        )
+        onPauseOrDispose {
+            player?.release()
+            player = null
+        }
+    }
+    // Prepare media
+    LaunchedEffect(player, assetUrl) {
+        if (player != null && assetUrl != null) {
+            player?.setMediaItem(MediaItem.fromUri(assetUrl))
+            player?.prepare()
+            player?.playWhenReady = playWhenReady
+        }
+    }
+    // Draw player
+    player?.let { preparedPlayer ->
+        Box(
+            modifier = modifier,
+            contentAlignment = Alignment.Center,
+        ) {
+            // Video/Audio player
+            AndroidView(
+                modifier = Modifier
+                    .matchParentSize()
+                    .background(Color.Black),
+                factory = {
+                    val playerView = createPlayerView(it, preparedPlayer)
+                    if (showControllerInitially) playerView.showController()
+                    playerView
+                },
+            )
+            // Thumbnail
+            if (thumbnailEnabled && showThumbnail) {
+                VideoThumbnail(
+                    modifier = Modifier.matchParentSize(),
+                    thumbnailUrl = thumbnailUrl,
+                    showPlayButton = showPlayButton,
+                    onPlayClick = {
+                        showThumbnail = false
+                        showPlayButton = false
+                        preparedPlayer.play()
+                    },
+                )
+            }
+            // Buffering indicator
+            if (showBuffering && !showThumbnail) {
+                LoadingIndicator()
+            }
+        }
+    }
+}
+
+/**
+ * A composable that displays a thumbnail image for a video with an optional play button overlay.
+ *
+ * This component shows a video thumbnail image with a centered play button that can be clicked
+ * to start video playback. The thumbnail image is only displayed if video thumbnails are
+ * enabled in the ChatTheme configuration.
+ *
+ * When clicked, the component triggers the provided [onPlayClick] callback to begin video playback.
+ * The thumbnail has a black background to ensure visual consistency when no thumbnail image
+ * is available or while it's loading.
+ *
+ * @param thumbnailUrl The URL of the thumbnail image to display, or null if no thumbnail is available.
+ * @param showPlayButton Whether to display the play button overlay on top of the thumbnail.
+ * @param onPlayClick Callback invoked when the user clicks on the thumbnail or play button to start playback.
+ * @param modifier Modifier to be applied to the thumbnail container.
+ */
+@Composable
+private fun VideoThumbnail(
+    thumbnailUrl: String?,
+    showPlayButton: Boolean,
+    onPlayClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val data = if (ChatTheme.videoThumbnailsEnabled) {
```

Comment:
Maybe this verification could be done by the composable that calls `VideoThumbnail` so this entire composable wouldn't be called if `ChatTheme.videoThumbnailsEnabled` is false.
WDYT?